### ISSUE TITLE
How to do k-max pooling with proper dim ?

### ISSUE THREAD

**Issue by lethienhoa** [tensorflow/tensorflow#7293]
Hi folks,

Does anybody has this same issue as me ? I want to apply a k-max pooling to this var, however it's is not in the proper dimension yet:

For ex:
h = tf.Variable(batch_size,1,18,512)     # This is the obligatory dim's result I got when I apply Deep CNN in NLP

I want to have the result of k-max pooling like this:
h_kmaxpooling with dim (batch_size,1,**8**,512) 

If I use tf.nn.top_k(h, 8), it will result in (batch_size,1,18,**8**) instead.
I've tried **tf.transpose(x, perm=[batch_size, 1, 512, 18])** but it always give error in batch_size since it doesn't know how many examples are. Or may be I've set wrong tf.transpose for this case ?

**Comment by girving** [comment]
This question would be better asked on StackOverflow.  Github issues are for bug reports and feature requests.
